### DISCUSSION TITLE
Randomize port for QuarkusDevModeTest

### DISCUSSION THREAD

**Discussion by stephan-strate** [quarkusio/quarkus#40816] [Q&A]
I want to test my extension using QuarkusDevModeTest. I randomized the port using the application.properties.

Problem now is, that I can not access the application using RestAssured, because the randomized port is not accessible for me.

Is there another way? Did I miss something?

```
    @RegisterExtension
    final static QuarkusDevModeTest test = new QuarkusDevModeTest()
            .withApplicationRoot((jar) -> jar
                    .addAsResource(new StringAsset("quarkus.http.port=0"), "application.properties");
```

**Comment by maxandersen** [comment]
Try: quarkus.http.test.port

**Reply by cescoffier** [reply]
Might be `quarkus.http.test-port`.

**Reply by stephan-strate** [reply]
`quarkus.http.test-port` did not work.
The port did not randomize. I guess because the property does not work for dev mode?

**Reply by cescoffier** [reply]
Ah on dev mode it's quarkus.http.port. Set it to -1.

**Reply by stephan-strate** [reply]
I did set it to `0` already. Port is randomized, but RestAssured is not populated with it. So I can not access the application.

**Reply by cescoffier** [reply]
Are you in dev or test? You cannot use RestAssured in dev mode. 

In dev mode, you need `quarkus.http.port` set to 0 or -1.
In test mode, you need `quarkus.http.test-port` set to 0 or -1.

**Reply by maxandersen** [reply]
Maybe a reproducer is best? I have a feeling you are doing something we don't expect.

**Reply by stephan-strate** [reply]
Sure. The `GreetingDevModeTest` reproduces my problem.

Scenario 1: `quarkus.http.port=8080`
Test works. Port is blocked on testing agent.

Scenario 2: `quarkus.http.port=0`
Test does not work, because RestAssured does not know the randomized port. Port is not blocked on testing agent.

I would like to be able to have the port randomized and still have access to it using RestAssured.

[code-with-quarkus.zip](https://github.com/user-attachments/files/15544976/code-with-quarkus.zip)

**Reply by maxandersen** [reply]
I get this error when running `mvn package` - which is expected but not the error you see. how are you expecting to run this?

`[ERROR]   GreetingDevModeTest.testDevMode » IllegalState Cannot mix both io.quarkus.test.QuarkusDevModeTest based tests and io.quarkus.test.junit.QuarkusTestExtension based tests in the same run`

if i delete the other tests and use `quarkus.http.test.port=0` it works?

**Reply by stephan-strate** [reply]
Yeah, thats right. Did not execute all tests, thats why I missed that in the reproducer. Normally I am testing an extension.

It works, but the dev mode is started using port 8080 and is not randomized.

**Comment by radcortez** [comment]
Fixed by https://github.com/quarkusio/quarkus/pull/55016
